sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala

```diff
@@ -89,6 +94,23 @@ class UserDefinedTypeSuite extends QueryTest with SharedSQLContext with ParquetT
     assert(featuresArrays.contains(new MyDenseVector(Array(0.2, 2.0))))
   }
 
+  test("user type with ScalaReflection") {
+    val points = Seq(
+      MyLabeledPoint(1.0, new MyDenseVector(Array(0.1, 1.0))),
+      MyLabeledPoint(0.0, new MyDenseVector(Array(0.2, 2.0))))
+
+    val schema = ScalaReflection.schemaFor[MyLabeledPoint].dataType.asInstanceOf[StructType]
+    val attributeSeq = schema.toAttributes
+
+    val pointEncoder = encoderFor[MyLabeledPoint]
+    val unsafeRows = points.map(pointEncoder.toRow(_).copy())
```

Comment:
can we also test `encoder.fromRow`?

we can just create a `MyLabelPoint` and encode it to `InternalRow` and decode it back by encoder, and check if the decoded `MyLabelPoint` is same with the original one.